### PR TITLE
[KAPT] KT-29929 Keep constant references in annotations with array values

### PR DESCRIPTION
**Pull request by zveznicht**

### CODE REVIEW

File: plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt [hunk 1]
```diff
@@ -1268,6 +1260,37 @@ class ClassFileToSourceStubConverter(val kaptContext: KaptContextForStubGenerati
             return treeMaker.Select(typeExpression, treeMaker.name("class"))
         }
 
+        fun unwrapArgumentExpression() =
```
Comment (yanex) [inline plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt:1263]: I suggest specifying return types for non-trivial functions.

File: plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt [hunk 2]
```diff
@@ -1268,6 +1260,37 @@ class ClassFileToSourceStubConverter(val kaptContext: KaptContextForStubGenerati
...
+            when (singleArg) {
+                is KtCallExpression -> {
+                    val resultingDescriptor = singleArg.getResolvedCall(kaptContext.bindingContext)?.resultingDescriptor
+
+                    if (resultingDescriptor is FunctionDescriptor && resultingDescriptor.fqNameSafe.asString() == "kotlin.arrayOf")
```
Comment (yanex) [inline plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt:1268]: Do we have a test with `arrayOf()`?

Comment (yanex) [reply plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt:1268]: Also, what about `intArrayOf()`, `booleanArrayOf()` and other similar functions?

Comment (yanex) [reply plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt:1268]: Also, you can avoid hard-coding the function fqn. All names are listed in `org.jetbrains.kotlin.resolve.ArrayFqNames`.

Comment (zveznicht) [reply plugins/kapt3/kapt3-compiler/src/org/jetbrains/kotlin/kapt3/stubs/ClassFileToSourceStubConverter.kt:1268]: I just extracted function from existing code ;). But ok, I'll have a look and fix

**Comment by zveznicht** [comment]
Merged in https://github.com/JetBrains/kotlin/commit/8071a1e246758fb3af5120d70faf91725b5e583d

**Pull request closed without merging**